src/AdoNetCore.AseClient/Internal/TypeMap.cs

```diff
@@ -37,27 +37,28 @@ internal static class TypeMap
+        private static readonly Dictionary<Type, AseDbType> NetTypeToAseDbTypeMap = new Dictionary<Type, AseDbType>
         {
-            {typeof(bool), (value, length) => TdsDataType.TDS_BIT},
-            {typeof(byte), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT1},
-            {typeof(sbyte), (value, length) => TdsDataType.TDS_INTN},
-            {typeof(short), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT2},
-            {typeof(ushort), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT2},
-            {typeof(int), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT4},
-            {typeof(uint), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT4},
-            {typeof(long), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT8},
-            {typeof(ulong), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT8},
-            {typeof(char), (value, length) => TdsDataType.TDS_CHAR},
-            {typeof(char[]), (value, length) => TdsDataType.TDS_CHAR},
-            {typeof(string), (value, length) => TdsDataType.TDS_LONGBINARY},
-            {typeof(byte[]), (value, length) => length <= VarLongBoundary ? TdsDataType.TDS_BINARY : TdsDataType.TDS_LONGBINARY},
-            {typeof(Guid), (value, length) => TdsDataType.TDS_BINARY},
-            {typeof(decimal), (value, length) => TdsDataType.TDS_NUMN},
-            {typeof(AseDecimal), (value, length) => TdsDataType.TDS_NUMN},
-            {typeof(float), (value, length) => value == DBNull.Value ? TdsDataType.TDS_FLTN : TdsDataType.TDS_FLT4},
-            {typeof(double), (value, length) => value == DBNull.Value ? TdsDataType.TDS_FLTN : TdsDataType.TDS_FLT8},
-            {typeof(DateTime), (value, length) => TdsDataType.TDS_BIGDATETIMEN }
+            {typeof(bool), AseDbType.Bit},
+            {typeof(byte), AseDbType.TinyInt},
+            {typeof(sbyte), AseDbType.SmallInt},
+            {typeof(short), AseDbType.SmallInt},
+            {typeof(ushort), AseDbType.UnsignedSmallInt},
+            {typeof(int), AseDbType.Integer},
+            {typeof(uint), AseDbType.UnsignedInt},
+            {typeof(long), AseDbType.BigInt},
+            {typeof(ulong), AseDbType.UnsignedBigInt},
+            {typeof(char), AseDbType.VarChar},
+            {typeof(char[]), AseDbType.VarChar},
+            {typeof(string), AseDbType.VarChar},
+            {typeof(byte[]), AseDbType.VarBinary},
+            {typeof(Guid), AseDbType.VarBinary},
+            {typeof(decimal), AseDbType.Decimal},
+            {typeof(AseDecimal), AseDbType.Decimal},
+            {typeof(float), AseDbType.Real},
+            {typeof(double), AseDbType.Double},
+            {typeof(DateTime), AseDbType.DateTime},
+            {typeof(TimeSpan), AseDbType.DateTime}
```

Comment:
This was a gap?

Reply:
Not a gap -- reference driver doesn't support `TimeSpan` as input (or output), but we can.